pkg/storage/secret/metadata/outbox_store_test.go

```diff
@@ -0,0 +1,225 @@
+	"github.com/stretchr/testify/require"
+)
+
+type outboxStoreModel struct {
+	rows []contracts.OutboxMessage
+}
+
+func newOutboxStoreModel() *outboxStoreModel {
+	return &outboxStoreModel{}
+}
+
+func (model *outboxStoreModel) Append(messageID string, message contracts.AppendOutboxMessage) {
+	model.rows = append(model.rows, contracts.OutboxMessage{
+		Type:            message.Type,
+		MessageID:       messageID,
+		Name:            message.Name,
+		Namespace:       message.Namespace,
+		EncryptedSecret: message.EncryptedSecret,
+		KeeperName:      message.KeeperName,
+		ExternalID:      message.ExternalID,
+	})
+}
+
+func (model *outboxStoreModel) ReceiveN(n uint) []contracts.OutboxMessage {
+	maxMessages := min(len(model.rows), int(n))
+	if maxMessages == 0 {
+		return []contracts.OutboxMessage{}
+	}
+	return model.rows[:maxMessages]
+}
+
+func (model *outboxStoreModel) Delete(messageID string) {
+	oldLen := len(model.rows)
+	model.rows = slices.DeleteFunc(model.rows, func(m contracts.OutboxMessage) bool {
+		return m.MessageID == messageID
+	})
+	if len(model.rows) != oldLen-1 {
+		panic("Delete: deleted more than one message")
+	}
+}
+
+func TestOutboxStoreModel(t *testing.T) {
+	t.Parallel()
+
+	model := newOutboxStoreModel()
+
+	require.Empty(t, model.ReceiveN(10))
+
+	appendOutboxMessage := contracts.AppendOutboxMessage{
+		Type:            contracts.CreateSecretOutboxMessage,
+		Name:            "s-1",
+		Namespace:       "n-1",
+		EncryptedSecret: secretv0alpha1.NewExposedSecureValue("value"),
+		KeeperName:      contracts.DefaultSQLKeeper,
+		ExternalID:      nil,
+	}
+
+	outboxMessage1 := contracts.OutboxMessage{
+		MessageID:       "message_id_1",
+		Type:            contracts.CreateSecretOutboxMessage,
+		Name:            "s-1",
+		Namespace:       "n-1",
+		EncryptedSecret: secretv0alpha1.NewExposedSecureValue("value"),
+		KeeperName:      contracts.DefaultSQLKeeper,
+		ExternalID:      nil,
+	}
+
+	outboxMessage2 := contracts.OutboxMessage{
+		MessageID:       "message_id_2",
+		Type:            contracts.CreateSecretOutboxMessage,
+		Name:            "s-1",
+		Namespace:       "n-1",
+		EncryptedSecret: secretv0alpha1.NewExposedSecureValue("value"),
+		KeeperName:      contracts.DefaultSQLKeeper,
+		ExternalID:      nil,
+	}
+
+	model.Append("message_id_1", appendOutboxMessage)
+
+	require.Equal(t, []contracts.OutboxMessage{outboxMessage1}, model.ReceiveN(10))
+
+	model.Append("message_id_2", appendOutboxMessage)
+
+	require.Equal(t, []contracts.OutboxMessage{outboxMessage1, outboxMessage2}, model.ReceiveN(10))
+
+	model.Delete(outboxMessage1.MessageID)
+
+	require.Equal(t, []contracts.OutboxMessage{outboxMessage2}, model.ReceiveN(5))
+
+	model.Delete(outboxMessage2.MessageID)
+
+	require.Empty(t, model.ReceiveN(1))
+}
+
+func TestOutboxStore(t *testing.T) {
+	testDB := sqlstore.NewTestStore(t)
+	require.NoError(t, migrator.MigrateSecretSQL(testDB.GetEngine(), nil))
+
+	ctx := context.Background()
```

Comment:
Not telling you to change, but just something I learned recently, Go 1.24 added a `t.Context()` that can be handy for these situations.